src/lib/y2network/config_writer.rb

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+module Y2Network
+  module ConfigWriter
+    # Config writer for a given source
+    #
+    # @param source [Symbol] Source name (e.g., :sysconfig)
+    # @return [#config] Configuration writer from {Y2Network::ConfigReader}
+    def self.for(source)
```

Comment:
should be part of config be also type of network daemon? So we do not need to call stuff like `for(:wicked).write(config)` This looks a bit strange and especially as from my exp always there is few backend specific options in configuration ( e.g. in bootloader it is always the case ). So I think part of configuration should be also target backend.

Reply:
btw if you are interested how I did it in bootloader. I create bootloader factory that can read current backend, propose new backend or simply get config for given backend. See https://github.com/yast/yast-bootloader/blob/master/src/lib/bootloader/bootloader_factory.rb